Tasks/Gradle/sonarqube-common/sonarqube-server.ts

```diff
@@ -0,0 +1,112 @@
+import Q = require('q');
+import url = require('url');
+import https = require('https');
+import http = require('http');
+import {IncomingMessage} from 'http';
+
+import {SonarQubeEndpoint} from './sonarqube-common';
+import sqCommon = require('./sonarqube-common');
+
+import tl = require('vsts-task-lib/task');
+
+export interface ISonarQubeServer {
+    callSonarQubeRestEndpoint(path:string):Q.Promise<Object>;
+
+    getEndpoint():SonarQubeEndpoint;
+}
+
+export class SonarQubeServer implements ISonarQubeServer {
+
+    private endpoint:SonarQubeEndpoint;
+
+    constructor() {
+        this.endpoint = sqCommon.getSonarQubeEndpoint();
```

Comment:
We have way too much logic in that soanrqube-common.ts file. To break it up, create a separate file for the endpoint and move all the logic there. It's not related to your work, but http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule